app/src/main/java/de/tum/in/tumcampusapp/component/ui/news/TopNewsStore.kt

```diff
@@ -60,7 +61,10 @@ class RealTopNewsStore @Inject constructor(
         val showTopNews = sharedPrefs.getBoolean(CardManager.SHOW_TOP_NEWS, true)
 
         val displayUntil = sharedPrefs.getString(Const.NEWS_ALERT_SHOW_UNTIL, "")
```

Comment:
We should rather add an extension method to `SharedPreferences` that can return `null`. Then, we can singly perform a null check instead of using a try-catch.

Reply:
That still does not make this code part any cleaner or nicer to look at. The best option would rather be to actually not trigger the display of the card, if there are no top news returned from the API. That seemed also a bit complex as in regard how the data is parsed and when the display of the card is checked. 

Feel free to push to this branch.

Reply:
Actually a 404 causes an exception, that I did not know how to deal with, as it would not allow to be passed on to the Store, but would fail in the repository.

I've added some better logic to handle an empty response from the backend.